lib/tasks/migrate.rake

```diff
@@ -261,6 +261,32 @@ namespace :migrate do
     end
   end
 
+  namespace :media do
+    # Media resources id 1 - 457799 are all subjects for project Snapshot Supernova.
+    # They currently exist outside of our media folder organizational structure.
+    # All files have already been copied over to /subject_location, this task must be run
+    # to update the media record src (which points to where the file is located)
+    desc 'Update src to subject_locations for Snapshot Supernova media'
+    task rewrite_supernova_file_locations: :environment do
+      # this will return match data that will give us just the file name,
+      # which is the part of the string following the old path
+      OLD_PATH_REGEX = /\A(panoptes-uploads.zooniverse.org\/1\/0\/)?(.+)\z/.freeze
+
+      puts 'starting supernova src location rewrite'
+      Media.where('id <= 457799').find_each do |medium|
+        matches = OLD_PATH_REGEX.match(medium.src)
+        # array index 1 will be nil if src does not start with panoptes-uploads.../0/
+        if matches[1]
+          # array index 2 is the second capture group (anything after panoptes-uploads.../0/)
+          file_name = matches[2]
+          new_path = 'subject_location/' + file_name
+          medium.update_column(:src, new_path)
+        end
+      end
```

Comment:
I'd put some logging code in here on each 1000 records to get some feedback on the rake task.
```suggestion
        puts "Processed 1000 records" if index % 1000 == 0 
      end
```

Adding the above will require an addition to the `find_each`
```
Media.where('id <= 457799').find_each.with_index do |medium, index|
```